protos/flyteidl/event/event.proto

```diff
@@ -150,4 +150,20 @@ message TaskExecutionEvent {
     // that should be recorded regardless of the lack of phase change.
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
+
+    // Metadata around how a task was executed.
+    TaskExecutionMetadata metadata        = 16;
+}
+
+// Holds metadata around how a task was executed.
+message TaskExecutionMetadata {
+    // Includes the type of machine used for this specific task execution. 
+    enum InstanceType {
+        // The default instance type configured for the flyte application platform.
+        DEFAULT = 0; 
+
+        // The instance type configured for interruptible tasks.
+        INTERRUPTIBLE = 1;
+    }
+    InstanceType instance_type = 1;
```

Comment:
Is it called "Instance Type" does that not refer to the `p316xlarge` stuff?

Reply:
perhaps InstanceClass would be more appropriate (and less specific?) if it's preferable to use the p316xlarge stuff, then the freeform map may make more sense here too